src/components/RichText/RichTextField.tsx

```diff
@@ -234,7 +248,16 @@ export function RichTextField({
                       />
 
                       <FormHelperText id={`${id}-helper-text`}>
-                        {getHelperText(meta, helperText)}
+                        {getHelperText(
+                          meta,
+                          helperText ??
+                            (showCharacterCount && val ? (
+                              <>
+                                <FormattedNumber value={val.characterCount} />{' '}
+                                characters
+                              </>
+                            ) : undefined)
```

Comment:
This was a first pass. It might be good to right align. It might be good to separate out from actual helper text so both can exist at once.

Reply:
See my comments above 😄